src/components/transactions/MigrateV3/MigrateV3Modal.tsx

```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import { BasicModal } from 'src/components/primitives/BasicModal';
+import { ModalType, useModalContext } from 'src/hooks/useModal';
+
+import { MigrateV3ModalContent } from './MigrateV3ModalContent';
+
+export const MigrateV3Modal = () => {
+  const { type, close } = useModalContext();
+
+  return (
+    <BasicModal open={type === ModalType.V3Migration} setOpen={close}>
+      <MigrateV3ModalContent />
```

Comment:
I think I didn't understanad that part properly, will check

Reply:
Now I think I do understand, but it's again the same issue, why do we have wrong network state coupled with underlying asset and what is underlying asset in that case? If I'm reading correctly I need to adapt underlyingAsset to have an array instead of a single one, while it would be way simplier and usefull to expose switching network globally. Switching network and fetching balances are not exactly part of the same logic at least for migration. 
Thiking how to adapty it